### PR TITLE
chore: Add unit tests for Connector, including required mocks and stubs.

### LINKED ISSUE #401: Add unit tests for Connector
* Mock the AlloyDB Admin RPC
* Add a fake SSL Server
* Add unit tests to validate the Connector

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #401

### CODE REVIEW

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  @After
+  public void tearDown() throws Exception {
+    defaultExecutor.shutdownNow();
+  }
+
+  @Test
+  public void create_successfulPrivateConnection()
+      throws IOException, InterruptedException, NoSuchAlgorithmException, InvalidKeySpecException {
+    FakeSslServer sslServer = new FakeSslServer(SERVER_MESSAGE);
+    sslServer.start(PRIVATE_IP);
```
Comment (enocom) [inline alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java:71]: Should we add a stop method?

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    ConnectionConfig config =
+        new ConnectionConfig.Builder().withInstanceName(InstanceName.parse(INSTANCE_NAME)).build();
+
+    Connector connector = newConnector(config.getConnectorConfig());
```
Comment (enocom) [inline alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java:76]: nit: passing in the mock as a second argument makes the relationship between the fake SSL server and the connector clean in my opinion.

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/StubAlloyDBAdminClientFactory.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            .directExecutor()
+            .addService(mock)
+            .build()
+            .start();
+
+    grpcCleanup.register(server);
+
+    ExecutorProvider executorProvider =
+        InstantiatingExecutorProvider.newBuilder().setExecutorThreadCount(1).build();
+    ;
```
Comment (enocom) [inline alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/StubAlloyDBAdminClientFactory.java:54]: dev artifact?

**Review by enocom — APPROVED** [review]
:clap: :clap: :clap: Really nice and clean implementation.

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java (hunk 1 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java:71]: added.

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java (hunk 2 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/ConnectorTest.java]: updated.

File: alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/StubAlloyDBAdminClientFactory.java (hunk 3 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/test/java/com/google/cloud/alloydb/StubAlloyDBAdminClientFactory.java]: udpated.

**Pull request merged**